lib/player.js

```diff
@@ -4,96 +4,72 @@ function Player(name, x, y, mass, keyTracker) {
   this.y    = y;
   this.mass = mass || 5;
   this.keyTracker = keyTracker;
-
-  this.movePlayer = function(xOffset, yOffset) {
-    this.x += xOffset;
-    this.y += yOffset;
-  };
-  this.moveLeft = this.movePlayer.bind(this, -1, 0);
-  this.moveRight = this.movePlayer.bind(this, 1, 0);
-  this.moveUp = this.movePlayer.bind(this, 0, -1);
-  this.moveDown = this.movePlayer.bind(this, 0, 1);
```

Comment:
Let's keep the above partial application. Let's add 'speed' as an argument into the move function and move them by xOffset * speed and yOffset * speed. Let's do this by adding speed as an attribute on the player and defaulting it.